portal-kernel/test/unit/com/liferay/portal/kernel/util/LocaleUtilTest.java

```diff
@@ -130,6 +130,25 @@ public void testFromLanguageIdConsistency() {
 		Assert.assertNull(LocaleUtil.fromLanguageId("de_DE", true, false));
 	}
 
+	@Test
+	public void testFromLanguageIdLocaleIsCreatedAndRetrievableWhenNoValidationDone() {
+		LanguageUtil languageUtil = new LanguageUtil();
+
+		Language language = Mockito.mock(Language.class);
+
+		languageUtil.setLanguage(language);
+
+		Mockito.when(
+			language.isAvailableLocale(Locale.ITALY)
+		).thenReturn(
+			false
+		);
+
+		Assert.assertSame(
```

Comment:
@marianoalvarosaiz You commit message says "Ensure that a locale is returned if no validation is performed", but you test code did not assert a locale is returned. You only asserted you get same result when you invoked it twice, but you won't know if it is null or not.

Reply:
@marianoalvarosaiz Forget it, I fixed it in the later commits.

Tina